src/test/java/com/shapesecurity/shift/es2016/test262/Test262.java

```diff
@@ -0,0 +1,196 @@
+package com.shapesecurity.shift.es2016.test262;
+
+import com.shapesecurity.functional.data.ImmutableList;
+import com.shapesecurity.functional.data.ImmutableSet;
+import com.shapesecurity.shift.es2016.ast.Program;
+import com.shapesecurity.shift.es2016.parser.EarlyError;
+import com.shapesecurity.shift.es2016.parser.EarlyErrorChecker;
+import com.shapesecurity.shift.es2016.parser.JsError;
+import com.shapesecurity.shift.es2016.parser.Parser;
+import com.shapesecurity.shift.es2016.test262.expectations.XFailHelper;
+import org.junit.Assert;
+import org.junit.Test;
+import org.yaml.snakeyaml.Yaml;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class Test262 {
+
+	private static final Yaml yamlParser = new Yaml();
+
+	private static final String testsDir = "src/test/resources/test262/test/";
+
+	@Nullable
+	private static Test262Info extractTest262Info(@Nonnull String path, @Nonnull String source) {
+		// extract comment block
+		int test262CommentBegin = source.indexOf("/*---");
+		if (test262CommentBegin < 0) {
+			return null;
+		}
+		test262CommentBegin += 5;
+		int test262CommentEnd = source.indexOf("---*/", test262CommentBegin);
+		if (test262CommentEnd < 0) {
+			return null;
+		}
+		String yaml = source.substring(test262CommentBegin, test262CommentEnd);
+		Object rawParsedYaml = yamlParser.load(yaml);
+		if (!(rawParsedYaml instanceof Map)) {
+			return null;
+		}
+		Map<String, Object> parsedYaml = (Map<String, Object>) rawParsedYaml;
+		// extract flags and negative
+		Object rawNegative = parsedYaml.get("negative");
+		Test262Info.Test262Negative negativeEnum = Test262Info.Test262Negative.NONE;
+		if (rawNegative != null) {
+			if (!(rawNegative instanceof Map)) {
+				return null;
+			}
+			Map<String, Object> negative = (Map<String, Object>) rawNegative;
+			String phase = (String) negative.get("phase");
+			if (phase == null) {
+				return null;
+			}
+			switch (phase) {
+				case "parse":
+					negativeEnum = Test262Info.Test262Negative.PARSE;
+					break;
+				case "early":
+					negativeEnum = Test262Info.Test262Negative.EARLY;
+					break;
+				default:
```

Comment:
I don't like defaults here. We should throw if we encounter something we don't recognize.